core/controllers/library_test.py

```diff
@@ -194,14 +197,138 @@ def test_library_handler_for_created_explorations(self):
+        activity_summary = response_dict['activity_summary_dicts_by_category'][
+            0]['activity_summary_dicts']
+        self.assertEqual(
+            len(activity_summary), 1)
+        self.assertDictContainsSubset({
+            'id': '0',
+            'category': 'Welcome',
+            'title': 'Welcome to Oppia!',
+            'language_code': 'en',
+            'objective': 'become familiar with Oppia\'s capabilities',
+            'status': rights_manager.ACTIVITY_STATUS_PUBLIC,
+        }, activity_summary[0])
+
+    def test_library_index_handler_for_featured(self):
+        """Test the handler for featured scenario."""
+        response_dict = self.get_json(feconf.LIBRARY_INDEX_DATA_URL)
+        self.assertDictContainsSubset({
+            'activity_summary_dicts_by_category': [],
+            'preferred_language_codes': ['en'],
+        }, response_dict)
+
+        # Load a demo.
+        exp_services.load_demo('0')
+        exploration_ref = activity_domain.ActivityReference(
+            constants.ACTIVITY_TYPE_EXPLORATION, '0')
+        activity_services.update_featured_activity_references([exploration_ref])
+
+        response_dict = self.get_json(feconf.LIBRARY_INDEX_DATA_URL)
+        self.assertEqual(
+            len(response_dict['activity_summary_dicts_by_category']), 1)
+        self.assertDictContainsSubset({
+            'preferred_language_codes': ['en'],
+        }, response_dict)
+        self.assertDictContainsSubset({
+            'categories': [],
+            'header_i18n_id': (
+                feconf.LIBRARY_CATEGORY_FEATURED_ACTIVITIES),
+            'has_full_results_page': False,
+            'full_results_url': None,
+        }, response_dict['activity_summary_dicts_by_category'][0])
+        activity_summary = response_dict['activity_summary_dicts_by_category'][
+            0]['activity_summary_dicts']
+        self.assertEqual(
+            len(activity_summary), 1)
+        self.assertDictContainsSubset({
+            'id': '0',
+            'category': 'Welcome',
+            'title': 'Welcome to Oppia!',
+            'language_code': 'en',
+            'objective': 'become familiar with Oppia\'s capabilities',
+            'status': rights_manager.ACTIVITY_STATUS_PUBLIC,
+        }, activity_summary[0])
+
+
 class LibraryGroupPageTests(test_utils.GenericTestBase):
 
+    def setUp(self):
+        super(LibraryGroupPageTests, self).setUp()
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.viewer_id = self.get_user_id_from_email(self.VIEWER_EMAIL)
+
     def test_library_group_pages(self):
         """Test access to the top rated and recently published pages."""
         self.get_html_response(feconf.LIBRARY_TOP_RATED_URL)
 
         self.get_html_response(feconf.LIBRARY_RECENTLY_PUBLISHED_URL)
 
+    def test_library_group_page_for_user_preferred_language(self):
+        """Test the access to user's preferences."""
```

Comment:
In this test, I want to test if the handler gets the preference correctly when the user logs in. The reason I change the language is that the language preference would be "en" whether the user logs in or not. Could you give me some hints to improve it? I can see this test case is confusing.

Reply:
Ah, thanks for the clarification. It is fine to leave the test as-is then. However, I would recommend putting in a comment letting others know that we need to set a different preference because the default is `en`.

Reply:
[Fixed in this commit](https://github.com/oppia/oppia/pull/6760/commits/95f7a4b0a3268af9334af5256c10ad6a8bf17ae7)

Thank you!